float32[] x_values
float32[] y_values